server/welcomebot.go

```diff
@@ -113,6 +113,7 @@ func (p *Plugin) renderWelcomeMessage(messageTemplate MessageTemplate, configMes
 
 			for _, channelName := range configAction.ChannelsAddedTo {
 				p.joinChannel(action, channelName)
+				time.Sleep(time.Second)
```

Comment:
I'm in favour of doing it this way, as it should fix the issue out-of-the-box. A config setting seems cumbersome and confusing for admins. 

This code only runs once per user, i.e. on their creation. Having a delay is justifiable. Even with 60 channels (!), that is just one minute of wait time.

Reply:
Though I wonder if the user is pinged with a "ding" every second. I haven't actually seen this in action yet

Reply:
Dear Michael @mickmister and Ben @hanzei, thanks for providing this fix and your thoughts. I agree with the concerns, though. This wait-time as the only measure is rather unreliable. You can never guess an optimal wait-time for all cases. Sometimes you bother users with unexpected long wait-times, sometimes you bother them with duplicate categories. 

Suggestions:

1. Is there a way to determine if a category needs to be created, but creation has not yet been triggered (this might be the tricky part), check if the category already exists, wait-retry if is it not yet created but triggered, add member if it is already created? 

2. Is there a way to wait only if a category assignment is configured in the channel actions?

What do you think about these suggestions?

That's of course more challenging to implement, but you might have an idea how to achieve that. Thank you very much for taking these considerations into account!

Reply:
@mickmister Are you looking into this or we should ?

Reply:
@hanzei Note that there is already a property on the `WelcomeMessage` struct called `DelayInSeconds`, which allows the admin to set a delay between certain operations, so introducing another part of the config with this pattern would not be out of the ordinary. 0/5 but in this case it make sense to allow the admin to set this, so we don't have to revisit it if the wait we choose is not high enough.